PWGJE/TableProducer/emcalClusterHadronicCorrectionTask.cxx

```diff
@@ -0,0 +1,309 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+// **Hadronic Correction in the EMCAL framework: to avoid the double counting of the charged particles' contribution in jets**
+/// \author Archita Rani Dash <archita.rani.dash@cern.ch>
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <unordered_map>
+#include <cmath>
+#include <TF1.h>
+#include <vector>
+
+#include "CCDB/BasicCCDBManager.h"
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ASoA.h"
+#include "Framework/HistogramRegistry.h"
+
+#include "DetectorsBase/GeometryManager.h"
+
+#include "PWGJE/DataModel/EMCALClusters.h"
+#include "PWGJE/DataModel/EMCALMatchedTracks.h"
+#include "PWGJE/DataModel/EMCALMatchedCollisions.h"
+#include "PWGJE/DataModel/emcalClusterHadronicCorrectionTask.h"
+
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Common/DataModel/PIDResponse.h"
+#include "DataFormatsEMCAL/Cell.h"
+#include "DataFormatsEMCAL/Constants.h"
+#include "DataFormatsEMCAL/AnalysisCluster.h"
+#include "EMCALBase/Geometry.h"
+#include "EMCALBase/ClusterFactory.h"
+#include "EMCALBase/NonlinearityHandler.h"
+#include "EMCALReconstruction/Clusterizer.h"
+#include "PWGJE/Core/JetUtilities.h"
+#include "TVector2.h"
+
+#include "CommonDataFormat/InteractionRecord.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using collisionEvSelIt = o2::soa::Join<o2::aod::Collisions, o2::aod::EvSels>::iterator;
+using selectedClusters = o2::soa::Filtered<o2::aod::EMCALClusters>;
```

Comment:
one thing i would recommend here is not performing this on filtered clusters, otherwise this table would not be joinable with the original cluster table, unless exactly the same filters are used everywhere.

Reply:
Done